sklearn/conftest.py

```diff
@@ -276,14 +280,11 @@ def pytest_configure(config):
     if not config.pluginmanager.hasplugin("sklearn.tests.random_seed"):
         config.pluginmanager.register(random_seed)
 
-
-def pytest_collectstart(collector):
-    # XXX: Easiest way to ignore pandas Pyarrow DeprecationWarning in the
-    # short-term. See https://github.com/pandas-dev/pandas/issues/54466 for
-    # more details.
-    warnings.filterwarnings(
-        "ignore", message=r"\s*Pyarrow", category=DeprecationWarning
-    )
+    if environ.get("SKLEARN_WARNINGS_AS_ERRORS", "0") != "0":
+        # This seems like the only way to programmatically change the config
+        # filterwarnings. This was suggested in
+        # https://github.com/pytest-dev/pytest/issues/3311#issuecomment-373177592
+        config._inicache["filterwarnings"] += get_pytest_filterwarning_str()
```

Comment:
Good point although it is kind of not encouraged in https://github.com/pytest-dev/pytest/issues/3311#issuecomment-373177592

> while addinivalue_line exists, it feels it was added solely to expand the list of builtin markers, which seems like it could have been done in some other way.

Reply:
OK I used addinivalue_line, at least it is public ...